### PR TITLE
Add proper documentation for binding modes.

### LINKED ISSUE #1996: Userguide doesn't document modes properly
The userguide doesn't really document what modes are and how to use them, except for an "on the side" kind of example for the resizing configuration mode. This feature should be documented properly, and should also take #1994 into account once merged.

i3 4.11

### PR DESCRIPTION
**Pull request by Airblader**
fixes #1996

### CODE REVIEW

File: docs/userguide [hunk 1]
```diff
@@ -2001,8 +2056,6 @@ move window|container to mark <mark>
 for_window [instance="tabme"] move window to mark target
 --------------------------------------------------------
 
-[[resizingconfig]]
```
Comment (stapelberg) [inline docs/userguide]: Can you leave this here so as to not break user’s bookmarks?

File: docs/userguide [hunk 2]
```diff
@@ -441,6 +443,57 @@ bindsym button9 move left
...
+[[binding_modes]]
+
+=== Binding modes
+
+You can have multiple sets of bindings by using different binding modes. When
+you switch to another binding mode, all bindings from the current mode are
+released and only the bindings defined in the new mode are valid for as long as
+you stay in that binding mode. The only predefined binding mode is +default+,
+which is the mode i3 starts out with and to which all bindings not defined in a
+specific bnding mode belong.
```
Comment (stapelberg) [inline docs/userguide]: typo: s/bnding/binding/

File: docs/userguide [hunk 3]
```diff
@@ -441,6 +443,57 @@ bindsym button9 move left
...
+mode [--pango_markup] <name>
+
+# command
+mode <name>
+----------------------------
+
+*Example*:
+---------------------------------------------------
+set $mode_launcher Launch: [f]irefox [t]hunderbird
+bindsym $mod+o mode "$mode_launcher"
```
Comment (stapelberg) [inline docs/userguide]: Maybe add a comment above line 485 to explain how this mode can be used:

> Press $mod+o, followed by either f, t, Esc or Return to launch firefox, thunderbird or return to the default mode, respectively.

File: docs/userguide [hunk 4]
```diff
@@ -149,8 +149,9 @@ it does not yet exist.
 The easiest way to resize a container is by using the mouse: Grab the border
 and move it to the wanted size.
 
-See <<resizingconfig>> for how to configure i3 to be able to resize
-columns/rows with your keyboard.
+You can also use <<binding_modes>> to define a mode for resizing via the
+keyboard. To see an example for this, look at the default config provided by
```
Comment (stapelberg) [inline docs/userguide]: Can you make this a link to the default config?

File: docs/userguide [hunk 5]
```diff
@@ -177,7 +178,8 @@ around. By grabbing the borders and moving them you can resize the window. You
 can also do that by using the <<floating_modifier>>. Another way to resize
 floating windows using the mouse is to right-click on the titlebar and drag.
 
-For resizing floating windows with your keyboard, see <<resizingconfig>>.
+For resizing floating windows with your keyboard, see the resizing binding mode
+provided by the i3 default config.
```
Comment (stapelberg) [inline docs/userguide]: Can you make this a link to the default config?

File: docs/userguide [hunk 6]
```diff
@@ -1455,8 +1510,8 @@ bar {
 
 Specifies whether the current binding mode indicator should be shown or not.
 This is useful if you want to hide the workspace buttons but still be able
-to see the current binding mode indicator.
-For an example of a +mode+ definition, see <<resizingconfig>>.
+to see the current binding mode indicator. See <<binding_modes>> to learn what
+they are and how to use them.
```
Comment (stapelberg) [inline docs/userguide]: s/they/modes/

File: docs/userguide [hunk 7]
```diff
@@ -2023,38 +2076,11 @@ many percentage points a *tiling container* should be grown or shrunk (the
...
-        bindsym Down        resize grow height 10 px or 10 ppt
-        bindsym Up          resize shrink height 10 px or 10 ppt
-        bindsym Right       resize grow width 10 px or 10 ppt
-
-        # back to normal: Enter or Escape
-        bindsym Return mode "default"
-        bindsym Escape mode "default"
-}
+It is recommended to define bindings for resizing in a dedicated binding mode.
+See <<binding_modes>> and the example in the i3 default config for more
```
Comment (stapelberg) [inline docs/userguide]: Can you make this a link to the default config?

**Comment by Airblader** [comment]
@stapelberg Done.

**Pull request merged**